Add support inference on SYCL devices (#9800)
dmlc/xgboost: plugin/sycl/README.md

> # SYCL-based Algorithm for Tree Construction

Reviewer: Add copyright

Author: done

This plugin adds support of SYCL programming model for prediction algorithms to XGBoost.

## Usage
Specify the 'device' parameter as one of the following options to offload model training and inference on SYCL device.

### Algorithms
| device | Description |
| --- | --- |
sycl | use default sycl device  |
sycl:gpu | use default sycl gpu  |
sycl:cpu | use default sycl cpu  |
sycl:gpu:N | use sycl gpu number N |
sycl:cpu:N | use sycl cpu number N |

Python example:
```python
param['device'] = 'sycl:gpu:0'
```

## Dependencies

> Building the plugin requires Data Parallel C++ Compiler (https://www.intel.com/content/www/us/en/developer/tools/oneapi/dpc-compiler.html)

Reviewer: ```suggestion
To build the plugin, install [Intel® oneAPI DPC++/C++ Compiler](https://www.intel.com/content/www/us/en/developer/tools/oneapi/dpc-compiler.html).
```

> Building the plugin requires Data Parallel C++ Compiler (https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/dpc-compiler.html)

Reviewer 2: ```suggestion
Building the plugin requires Data Parallel C++ Compiler (https://www.intel.com/content/www/us/en/developer/tools/oneapi/dpc-compiler.html)
```

## Build

> From the command line on Linux starting from the xgboost directory:

Reviewer: ```suggestion
From the ``xgboost`` directory, run:
```

<!--
******************************************************************************
* Copyright by Contributors 2017-2023
*******************************************************************************/-->

# SYCL-based Algorithm for Tree Construction

> This plugin adds support of SYCL programming model for prediction algorithms to XGBoost.

Reviewer 2: Let's add https://www.intel.com/content/www/us/en/docs/oneapi/programming-guide reference here

Author: done

## Usage
Specify the 'device' parameter as described in the table below to offload model training and inference on SYCL device.

### Algorithms
| device | Description |
| --- | --- |
sycl | use default sycl device  |
sycl:gpu | use default sycl gpu  |

> sycl:cpu | use default sycl cpu  |

Reviewer 2: Should we add more explanations here on expectations? Like cpu at this point would launch sycl kernels on CPU device and this at this point functionally supported but no performance is expected in this case.

Author: added

sycl:gpu:N | use sycl gpu number N |
sycl:cpu:N | use sycl cpu number N |

Python example:
```python
param['device'] = 'sycl:gpu:0'
```

## Dependencies

> To build the plugin, install [Intel® oneAPI DPC++/C++ Compiler](https://www.intel.com/content/www/us/en/developer/tools/oneapi/dpc-compiler.html).

Reviewer 2: I think we also should mention that once build - for running plugin you have to resolve DPC compiler runtime + drivers

Author: done
